lib/resources.js

```diff
@@ -617,7 +695,16 @@ const buildElasticQueryForFilters = function (params) {
   var filterClausesWithPaths = []
 
   if (params.filters) {
-    filterClausesWithPaths = Object.keys(params.filters).map((prop) => {
+    // Add clauses for dateAfter / dateBefore filters, if used:
+    filterClausesWithPaths = filterClausesWithPaths.concat(
+      filterClausesForDateParams(params.filters)
+    )
+
+    // Collect those filters that use a simple term match
+    const simpleMatchFilters = Object.keys(params.filters)
+      .filter((k) => FILTER_CONFIG[k].operator === 'match')
```

Comment:
Why are we adding this filtering?

Reply:
Previously we handled all filters in a single `reduce`. This change acknowledges that some filters can not be handled in that simple `reduce`, that some filters need to be considered together (for example when `dateBefore` is used alone we need a very different query than when `dateBefore` and `dateAfter` are used together). So the line highlighted selects only those filters that are configured as 'match' filters (which are the class of filters that can be handled by the `reduce`.